gamemodes/terrortown/entities/weapons/weapon_tttbase.lua

```diff
@@ -659,6 +668,99 @@ if CLIENT then
 		self.fCurrentSysTime = SysTime()
 	end
 
+	---
+	-- Adds a custom view model.
+	-- @note Multiple view models can be added, they are all rendered at once
+	-- @param string identifier The name of the added view model
+	-- @param ModelData modelData The model data table
+	-- @realm client
+	function SWEP:AddCustomViewModel(identifier, modelData)
+		self.VElements = self.VElements or {}
+		self.VElements[identifier] = modelData
+
+		modelbuilder.CreateModel(self, self.VElements[identifier])
+	end
+
+	---
+	-- Adds a custom world model.
+	-- @note Multiple world models can be added, they are all rendered at once
+	-- @param string identifier The name of the added world model
+	-- @param ModelData modelData The model data table
+	-- @realm client
+	function SWEP:AddCustomWorldModel(identifier, modelData)
+		self.WElements = self.WElements or {}
```

Comment:
people that are used to using swepck are used to the velement/welement naming convention, this would make migrating output more difficult

Reply:
Agreed, I kept all names from the tool so that it is identical. Also that way our third person renderer for the UI is compatible with both my implementation and the basic implementation without me adding something

However I'm wondering if having the same variable names can lead to clashes of the two renderers, hm. The boomerang is working fine for me, but maybe some addons do some weird shit that could break here? I'm not 100% sure.

Reply:
Then please describe the workflow on how someone adds such models etc since im not familiar with the outputs or anything of that tool.

Reply:
I've also never used that tool, but isn't this already a good description:

https://github.com/TTT-2/TTT2/blob/0f6b4b325a3ebe82705efa6d14a6d88c08247459/lua/ttt2/libraries/modelbuilder.lua#L6-L23

Reply:
Yes but this does not care about the variable naming we use to just store these elements.
I don't think this is an issue.

Reply:
> Then please describe the workflow on how someone adds such models etc since im not familiar with the outputs or anything of that tool.

If they are named @TimGoll they manually try and error with those variables until it looks nice. If they use the linked tool, the tool gives them this exact table. Therefore I kept all variable names inside the table identical, so that it is as low friction as possible

Reply:
also a lot of existing TTT1 equipment is made using said tool, so it will make porting older stuff significantly easier

Reply:
I've thought about this for a while and I think it would be best to rename the variables `WElements` and `VElements`. This way we would be safe from clashes with external addons that implement the renderer from the tool.

The variable names inside the table should stay unchanged though. Otherwise porting over addons will be tricky for devs just like @EntranceJew said